### ISSUE TITLE
Adding to List of PhotoViewGalleryPageOptions

### ISSUE THREAD

**Issue by dandansoysauce** [renancaraujo/photo_view#78]
`setState() called after dispose()` is being thrown when I'm trying to add an item to a `List<PhotoViewGalleryPageOptions>`

**My Setup:**

``` dart
return PhotoViewGallery(
  pageOptions: List<PhotoViewGalleryPageOptions>
);
```
And I'm doing `setState()` on every `page change - onPageChanged()`

**Comment by renancaraujo** [comment]
I think we should use a `builder` pattern instead of a list.

**Comment by renancaraujo** [comment]
Added the builder constructor on #119 and published in the version `0.2.3`.
